Mathlib/Combinatorics/Enumerative/Partition/Glaisher.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2025 Weiyi Wang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Weiyi Wang
+-/
+import Mathlib.Combinatorics.Enumerative.Partition.GenFun
+import Mathlib.RingTheory.PowerSeries.NoZeroDivisors
+
+/-!
+# Glaisher's theorem
+
+This file proves Glaisher's theorem: the number of partitions of an integer $n$ into parts not
+divisible by $d$ is equal to the number of partitions in which no part is repeated $d$ or more
+times.
+
+## Main declarations
+* `Nat.Partition.card_restricted_eq_card_countRestricted`: Glaisher's theorem.
+
+## References
+https://en.wikipedia.org/wiki/Glaisher%27s_theorem
+-/
+
+variable (R) [TopologicalSpace R] [T2Space R]
+
+namespace PowerSeries.WithPiTopology
+variable [CommRing R]
+
+omit [T2Space R] in
+theorem multipliable_one_sub_X_pow : Multipliable fun n ↦ (1 : R⟦X⟧) - X ^ (n + 1) := by
```

Comment:
This one and the next look generic enough that they should be in another file, perhaps nearer WithPiTopology?